app/models/package_manager/base.rb

```diff
@@ -160,6 +160,20 @@ def self.update(name, sync_version: :all, force_sync_dependencies: false)
       finalize_db_project(db_project)
     end
 
+    # Override this in the subclass to fetch the raw data from the upstream
+    # package manager. This can be any arbitrary data, and will passed on to the
+    # mapping() method to get a standard shape of data.
+    def self.project(_name)
+      nil
+    end
+
+    # Override this in the subclass to map the raw data from project() to
+    # a Hash of data that we'll need to save Project and Version records.
+    # Use the PackageManager::MappingBuilder to create the Hash.
+    def self.mapping(_raw_project)
+      nil
+    end
+
```

Comment:
Is it required that subclasses implement these? We could raise with a more specific error if so.